src/components/TimeSelectInput/index.js

```diff
@@ -94,6 +100,7 @@ export default class TimeSelectInput extends Component {
 
     handleClickOutside() {
         const { inputFocusedIndex } = this.state;
+        this.handleBlur();
```

Comment:
this is causing the blur to get called every time for each input, this is not a correct behaviour since only focused element should be able to blur